py/picca/delta_extraction/expected_fluxes/dr16_expected_flux.py

```diff
@@ -625,184 +688,72 @@ def compute_var_stats(self, forests):
+        )
 
         self.logger.progress(" Mean quantities in observer-frame")
         self.logger.progress(
             " loglam    eta      var_lss  fudge    chi2     num_pix valid_fit")
         for index in range(self.num_bins_variance):
-            # pylint: disable-msg=cell-var-from-loop
-            # this function is defined differntly at each step of the loop
-            def chi2(eta, var_lss, fudge):
-                """Compute the chi2 of the fit of eta, var_lss, and fudge for a
-                wavelength bin
-
-                Arguments
-                ---------
-                eta: float
-                Correction factor to the contribution of the pipeline
-                estimate of the instrumental noise to the variance.
-
-                var_lss: float
-                Pixel variance due to the Large Scale Strucure
-
-                fudge: float
-                Fudge contribution to the pixel variance
-
-                Global arguments
-                ----------------
-                (defined only in the scope of function compute_var_stats):
-
-                var_delta: array of floats
-                Variance of the delta field
-
-                var2_delta: array of floats
-                Square of the variance of the delta field
-
-                index: int
-                Index with the selected wavelength bin
-
-                num_var_bins: int
-                Number of bins in which the pipeline variance values are split
-
-                var_pipe_values: array of floats
-                Value of the pipeline variance in pipeline variance bins
-
-                num_qso: array of ints
-                Number of quasars in each pipeline variance bin
-
-                Return
-                ------
-                chi2: float
-                The obtained chi2
-                """
-                variance = eta * var_pipe_values + var_lss + fudge * fudge_ref / var_pipe_values
-                chi2_contribution = (
-                    var_delta[index * num_var_bins:(index + 1) * num_var_bins] -
-                    variance)
-                weights = var2_delta[index * num_var_bins:(index + 1) *
-                                     num_var_bins]
-                w = num_qso[index * num_var_bins:(index + 1) *
-                            num_var_bins] > 100
-                return np.sum(chi2_contribution[w]**2 / weights[w])
-
-            minimizer = iminuit.Minuit(chi2,
+            leasts_squares.set_fit_bins(index)
+
+            minimizer = iminuit.Minuit(leasts_squares,
                                        name=("eta", "var_lss", "fudge"),
-                                       eta=1.,
-                                       var_lss=0.1,
-                                       fudge=1.)
+                                       eta=eta[index],
+                                       var_lss=var_lss[index],
+                                       fudge=fudge[index] / FUDGE_REF)
```

Comment:
When all three variables are floated, then yes, the initialization is the same as before. Otherwise, they are initialized to the actual read values. 
Not sure what you mean with respect to the limits/errors

Reply:
Wait, what is FUDGE_REF? Is this the user input function with a reference fudge factor? If so, I'd imagine that most of the time we would provide fudge=0, so I don't think dividing by fudge_ref is a good idea... Am I missing something? @cramirezpe , @iprafols - happy to chat in person tomorrow.